providers/implementations/asymciphers/rsa_enc.c

```diff
@@ -90,7 +94,10 @@ static int rsa_init(void *vprsactx, void *vrsa)
 {
     PROV_RSA_CTX *prsactx = (PROV_RSA_CTX *)vprsactx;
 
-    if (prsactx == NULL || vrsa == NULL || !RSA_up_ref(vrsa))
+    if (!provider_is_running()
```

Comment:
Does FIPS require all these tests?  I.e., we can't just test when doing the fetch or creating the context?  I think we should ask the lab.

Reply:
More.  All operations are meant to stop in the error state.

I'm not planning on adding these checks to the update calls -- because it could be a potential performance impact.  Deallocation will also remain possible in the error state.

TLS fetches and caches algorithms so the fetch isn't sufficient by itself.  Context creation and duplication as well might be workable but it's only final beyond these.